### PR TITLE
Docstrings for all epimodel types and generate_latent_infs plus named arguments for constructors

### PR DESCRIPTION
**Pull request by SamuelBrand1**
This is a draft PR for on going work on documenting `AbstractEpiModel` subtypes and methods that dispatch on them.

## Current focus and decisions:

- As recommended [here](https://docs.sciml.ai/SciMLStyle/stable/#documentation) I am focusing documentation at the highest/most abstract level. This means that I'm going to give the example usages and most extensive documentation to `generate_latent_infs` dispatch on generic `AbstractEpiModel`.
- I don't think the METHODLIST was useful in our doc template and has been removed.
- Example usage of a `DirectInfections` goes beyond just constructing the struct, but also its main usage with `generate_latent_infs`.
- Example usage is not a doc test; it has a complicated output that works ok with a spread of `testitem`s. However, a next step could be to combine as per discussed in #48. _I've readvised this... simple doctests will be included_.

This is aimed at addressing #137 which is part of #35 and #97 . It sets my current expectation of docstrings #135 and #126 .

## Extra activity

- When it is absolutely straightforward to create a default constructor with named arguments using `@kwdef` I am doing it, because this also makes example usage easier. This moves towards addressing #133 , but won't close this issue.
- Advantage of above is that default priors are added directly aiming to address #131 .

### CODE REVIEW

**Comment by SamuelBrand1** [comment]
I've done everything in the opening description _except_ for the example usage of the EpiModels and `generate_latent_infs` is **not a doctest**.

The reason for this is that the pattern of doc testing against an expected result didn't seem to meet what we want from the examples and might lead into complexity around different random seed implementation on different OS etc. An alternative approach which just checks the examples run didn't seem to do anything that our testing CI doesn't do anyway.

Happy to get pushed back on this.

File: EpiAware/src/epimodels/directinfections.jl [hunk 1]
```diff
@@ -1,9 +1,115 @@
...
+```
+
+where ``g`` is a transformation function and the unconstrained initial infections
+    ``\hat{I}_0`` are sampled from a prior distribution, `initialisation_prior` which must
+be supplied to the `DirectInfections` constructor. The default `initialisation_prior` is
+the standard Normal `Distributions.Normal()`.
+
+## Constructor
+
+`DirectInfections` can be constructed by passing an `EpiData` object and and subtype of
```
Comment (seabbs) [inline EpiAware/src/epimodels/directinfections.jl:20]: repeated and

Comment (seabbs) [inline EpiAware/src/epimodels/directinfections.jl:20]:
```suggestion
`DirectInfections` can be constructed by passing an `EpiData` object and subtype of
```

**Review by seabbs — APPROVED** [review]
This is nice. It doesn't look like you have implemented field or return information for functions or made use of `DocStringExtensions`. I think we need at least a clear return field but perhaps in another issue.

**Pull request merged**